packages/analytics-js-plugins/src/oneTrust/index.ts

```diff
@@ -28,20 +33,82 @@ const OneTrust = (): ExtensionPlugin => ({
       const allowedConsentIds = (window as any).OnetrustActiveGroups.split(',').filter(
         (n: string) => n,
       );
+      const allowedConsents: Record<string, string> = {};
       const deniedConsentIds: string[] = [];
 
       // Get the groups(cookie categorization), user has created in one trust account.
       const oneTrustAllGroupsInfo: OneTrustGroup[] = (window as any).OneTrust.GetDomainData()
         .Groups;
 
       oneTrustAllGroupsInfo.forEach((group: OneTrustGroup) => {
-        const { CustomGroupId } = group;
-        if (!allowedConsentIds.includes(CustomGroupId)) {
+        const { CustomGroupId, GroupName } = group;
+        if (allowedConsentIds.includes(CustomGroupId)) {
+          allowedConsents[CustomGroupId.toUpperCase()] = GroupName.toUpperCase();
```

Comment:
Yes to make sure that I converted both ID and name as uppercase. What are your expectations here?

Reply:
We must ensure the category names are case-insensitive in the consent managers' space. Otherwise, we'd treat `Marketing` and `marketing` as the same.
To be on the safer side, I wouldn't convert the casing.

Reply:
Adressed